references/classification/WildFire/torch/train.py

```diff
@@ -0,0 +1,311 @@
+        criterion (torch.nn.Module): criterion object
+        master_bar (fastprogress.MasterBar): master bar of training progress
+        epoch (int): current epoch index
+        scheduler (torch.optim._LRScheduler, optional): learning rate scheduler
+        device (str): device hosting tensor data
+    Returns:
+        batch_loss (float): latch batch loss
+    """
+
+    # Training
+    model.train()
+    loader_iter = iter(train_loader)
+    train_loss = 0
+    for _ in progress_bar(range(len(train_loader)), parent=master_bar):
+
+        x, target = next(loader_iter)
+        target = target.type(torch.LongTensor).squeeze()
+        # Work with tensors on GPU
+        if device.startswith('cuda'):
+            x, target = x.cuda(), target.cuda()
+
+        batch_loss = train_batch(model, x, target, optimizer, criterion)
+        train_loss += batch_loss
+        if scheduler:
+            scheduler.step()
+
+        master_bar.child.comment = f"Batch loss: {batch_loss:.4}"
+
+    train_loss /= len(train_loader)
+
+    return train_loss
+
+
+def evaluate(model, test_loader, criterion, device='cpu'):
+    """Evaluation a model on a dataloader
+    Args:
+        model (torch.nn.Module): model to train
+        train_loader (torch.utils.data.DataLoader): validation dataloader
+        criterion (torch.nn.Module): criterion object
+        device (str): device hosting tensor data
+    Returns:
+        val_loss (float): validation loss
+        acc (float): top1 accuracy
+    """
+    model.eval()
+    val_loss, correct, targets = 0, 0, 0
+    with torch.no_grad():
+        for x, target in test_loader:
+
+            target = target.type(torch.LongTensor).squeeze()
+            # Work with tensors on GPU
+            if device.startswith('cuda'):
+                x, target = x.cuda(), target.cuda()
+
+            # Forward + Backward & optimize
+            outputs = model.forward(x)
+            val_loss += criterion(outputs, target).item()
+            # Index of max log-probability
+            pred = outputs.max(1, keepdim=True)[1]
+            correct += pred.eq(target.view_as(pred)).sum().item()
+            targets += x.size(0)
+    val_loss /= len(test_loader)
+    acc = correct / targets
+
+    return val_loss, acc
+
+
+def main(args):
+
+    if args.deterministic:
+        set_seed(42)
+
+    # Set device
+    if args.device is None:
+        if torch.cuda.is_available():
+            args.device = 'cuda:0'
+        else:
+            args.device = 'cpu'
+
+    #Create Dataloaders
+
+    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+
+    train_transforms = transforms.Compose([
+        transforms.RandomResizedCrop(size=args.resize // 7, scale=(0.8, 1.0)),
+        transforms.RandomRotation(degrees=15),
+        transforms.ColorJitter(),
+        transforms.RandomHorizontalFlip(),
+        transforms.CenterCrop(size=args.resize),  # Image net standards
```

Comment:
Generally CenterCrop is used for eval but not train set. Having two crop transformations seems unnecessary. It might be better to stick with RandomResizedCrop but discard CenterCrop